src/plugins/mer/merhardwaredevicewizardpages.cpp

```diff
@@ -522,5 +533,194 @@ bool MerHardwareDeviceWizardSetupPage::validatePage()
+    mainLayout->addWidget(errorLabel);
+
+    auto keyLayout = new QHBoxLayout(this);
+    keyLayout->addWidget(new QLabel(tr("Public key:")));
+    keyLayout->addWidget(m_keyComboBox);
+    mainLayout->addLayout(keyLayout);
+
+    auto deployLayout = new QHBoxLayout(this);
+    deployLayout->addWidget(deployKeyButton);
+    deployLayout->addWidget(m_deployStateLabel);
+    deployLayout->addWidget(m_progressIndicator);
+    deployLayout->addStretch();
+    mainLayout->addLayout(deployLayout);
+
+    QList<GpgKeyInfo> availableKeys;
+    if (Sfdk::isGpgAvailable()) {
+        bool ok;
+        QString error;
+        execAsynchronous(std::tie(ok, availableKeys, error), Sfdk::availableGpgKeys);
+        if (!ok)
+            Core::MessageManager::write(error, Core::MessageManager::Flash);
+    } else {
+        errorLabel->setText(tr("GPG utility not found!"));
+        errorLabel->setStyleSheet(QLatin1String("color: \"yellow\""));
+        deployKeyButton->setEnabled(false);
+        m_keyComboBox->setEnabled(false);
+    }
+
+    auto signingUsersModel = new SigningUserListModel(availableKeys);
+    m_keyComboBox->setModel(signingUsersModel);
+    int index = signingUsersModel->getIndexBy(MerSettings::signingUser());
+    m_keyComboBox->setCurrentIndex(index);
+}
+
+void MerHardwareDeviceWizardPackageKeyDeploymentPage::setDevice(const MerHardwareDevice::Ptr &device)
+{
+    m_device = device;
+}
+
+void MerHardwareDeviceWizardPackageKeyDeploymentPage::deployKey()
+{
+    QTC_ASSERT(m_device, return);
+
+    m_progressIndicator->show();
+    m_deployStateLabel->hide();
+
+    auto onDeployFinished = [&](bool ok) {
+        m_progressIndicator->hide();
+
+        m_deployStateLabel->setPixmap(ok ? Icons::OK.pixmap() : Icons::BROKEN.pixmap());
+        m_deployStateLabel->show();
+    };
+
+    const int index = m_keyComboBox->currentIndex();
+    const QModelIndex currentModelIndex = m_keyComboBox->model()->index(index, 0);
+    if (!currentModelIndex.isValid()) {
+        onDeployFinished(false);
+        return;
+    }
+    const QVariant signingUserVariant = m_keyComboBox->model()->data(currentModelIndex, Qt::UserRole);
+    const auto signingUser = signingUserVariant.value<GpgKeyInfo>();
+
+    MerHardwareDeviceGpgKeyDeploymentDialog dlg(m_device, signingUser.fingerprint, this);
+    onDeployFinished(dlg.exec() == QDialog::Accepted);
+}
+
+/*!
+ * \class PackageKeyDeployerPrivate
+ * \internal
+ */
+
+class PackageKeyDeployerPrivate : public QObject
+{
+    Q_OBJECT
+public:
+    PackageKeyDeployerPrivate() = default;
+
+    void deploy(const MerHardwareDevice::Ptr &device, const QString &id)
+    {
+        QTC_ASSERT(device, emit deployFinished(false, tr("No device set.")); return );
+
+        bool ok;
+        QString error;
+        execAsynchronous(std::tie(ok, error),
+                         &HardwareDevice::importPublicGpgKey,
+                         device->sshParameters(),
+                         id);
```

Comment:
Now I realized maybe we could create a temporary `Sfdk::HardwareDevice` instance just for the wizard purposes. I do not remember the details, so could you please try creating it like in `MerHardwareDeviceManager::onDeviceAddedOrUpdated` (just omit the `Sdk::addDevice` call) make `importPublicGpgKey` non-static and call it on the temporary instance? (If this works, it allows us moving more code from the wizard to the library easily, so it would be very nice! :))